haystack/pipelines/base.py

```diff
@@ -405,14 +405,19 @@ def eval(
             if params is None:
                 params = {}
             params["add_isolated_node_eval"] = True
-        queries = [label.query for label in labels]
-        for query, label in zip(queries, labels):
-            predictions = self.run(query=query, labels=label, params=params, debug=True)
+        for label in labels:
+            if label.filters is not None:
+                if params is None:
+                    params = {"filters": label.filters}
+                else:
+                    # join both filters and overwrite filters in params with filters in labels 
+                    params["filters"] = {**params.get("filters", {}), **label.filters}
```

Comment:
`params` or more precise `params["filters"]` seem to accumulate over multiple labels here. I think, we need to create to deepcopy global `params` for each label to prevent that.

Reply:
Agreed. 👍 Done.